### PR TITLE
feat: Improve routing UX

### PR DESCRIPTION
**Pull request by aomi**
Closes issue #39 by improving the routing UX doing the following:
- Shorten the URLs / routes as a whole.
- Remove spaces in the categories URLs.

Also, refactors some code reduce duplication, variable allocation and component prop organization. Notably the categories data is now unified under a single place.

TODO:
- Need to generate slugs for each club / course union.
- 404 page

### CODE REVIEW

**Comment by malcolmseyd** [comment]
Your TODO about generating slugs definitely needs to be addressed. The current slugging behavior is incomplete and causes breaking across the app.

Here's an example. If I go to the sports category, the path is `/categories/sports-and-recreation`. If I click on the poker club, the path becomes `/categories/Sports%20&%20Recreation/UVic%20Poker%20Club`. This is inconsistent and shows that #39 isn't properly addressed.

There's also a bug present. If I'm on the info page for the poker club and I click on the "Sports & Recreation" in the `PathItem`, it will redirect me to `/categories/sports%20&%20recreation`, which isn't a valid slug and causes a 404. This occurs because instead of using the correct slug, it's generated by downcasing the `clubType`, which doesn't always work.

In summary, slugs need to be generated before this can be merged.

**Comment by aomi** [comment]
@malcolmseyd Generated slugs are now being used :+1:

**Review by markksr5 — APPROVED** [review]
